File: transafrica_payroll/transafrica_payroll/report/consolidated_accounts_balances/consolidated_accounts_balances.py
# Copyright (c) 2022, Christopher Njogu and contributors
# For license information, please see license.txt

import frappe
from frappe import _
import erpnext
from erpnext.accounts.utils import get_balance_on


def execute(filters=None):
	columns = get_columns()
	data = []
	parent_account = filters.get("account")
	as_of_date = filters.get("as_of_date")
	accounts = frappe.db.get_list('Account', filters={'parent_account': parent_account}, fields=['name'])
	for account in accounts:
		data.append({
			"account": account.name,
			"balance": get_balance_on(account.name, as_of_date),
			"dr_cr": "Cr" if get_balance_on(account.name) < 0 else "Dr"
		})
	return columns, data


def get_columns():
	columns = [
		{
			'label': _('Account Name'),
			'fieldname': 'account',
			'options': 'Account',
			'width': 400
		},
		{
			'label': _('Balance'),
			'fieldname': 'balance',
			'fieldtype': 'Currency',
			'width': 260
		},
		{
			'label': _('Debit/Credit'),
			'fieldname': 'dr_cr',
			'fieldtype': 'Data',
			'width': 150
		}
	]

	return columns

File: transafrica_payroll/transafrica_payroll/report/consolidated_accounts_balances/consolidated_accounts_balances.js
// Copyright (c) 2022, Christopher Njogu and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Consolidated Accounts Balances"] = {
	"filters": [
		{
			"fieldname":"company",
			"label": __("Company"),
			"fieldtype": "Link",
			"options": "Company",
			"reqd": 1,
			"default": frappe.defaults.get_user_default("Company")
		},
		{
			"fieldname":"account",
			"label": __("Group Account"),
			"fieldtype": "Link",
			"options": "Account",
			"reqd": 1,
			"get_query": function() {
				var company = frappe.query_report.get_filter_value('company')
				return {
					"query": "erpnext.controllers.queries.get_account_list",
					"filters": [
						// ['Account', 'account_type', 'in', 'Bank, Cash'],
						['Account', 'is_group', '=', 1],
						['Account', 'disabled', '=', 0],
						['Account', 'company', '=', company],
					]
				}
			}
		},
		{
			"fieldname":"as_of_date",
			"label": __("Date"),
			"fieldtype": "Date",
			"default": frappe.datetime.get_today(),
			"reqd": 1
		}
	]
};
